### ISSUE TITLE
Instantiating RemoteWebDriver fails and Selenium.WebDriver 3.10.0 + doesn't use default system proxy

### ISSUE THREAD

**Issue by gschultz77** [SeleniumHQ/selenium#5772]
## Meta
OS: Windows 10
Selenium Version: Selenium.WebDriver 3.10.0 or 3.11.0 or 3.11.1 -  C#, .NET 4.6.1
Browser: RemoteWebDriver

Browser Version: na

## Expected Behavior
Attempting to instantiate a new RemoteWebDriver instance behind a corporate proxy (which is the system default proxy) should work, given that the proxy allows that traffic.
This works fine with Selenium.WebDriver 3.9.1 (and previous versions)

## Actual Behavior
Attempting to instantiate a new RemoteWebDriver with 3.10.0 and above fails - it appears that the request does **not** go via the proxy as previously.

Looking at issue #4478 I've tried to implement the code as provided by Jim that will implement a BeforeRemoteHttpRequest event, but the method that hooks into that event never gets hit (my code is identical to that provided in #4478) 

## Steps to reproduce
Even if you explicitly set the proxy configuration in App.Config like so, this appears to be ignored whereas 3.9.1 will use this configuration.

```
<system.net> 
    <defaultProxy enabled="true" useDefaultCredentials="true">
      <proxy proxyaddress="<proxy_url>"  />
    </defaultProxy>
</system.net>
```

**Comment by jimevans** [comment]
What happens if you use `WebRequest.GetSystemWebProxy()` instead of `WebRequest.DefaultWebProxy` in your event handler?

**Comment by gschultz77** [comment]
Same end result - fails, but the thing I don't understand is why the method attached to the BeforeRemoteHttpRequest event doesn't fire at all (debugging teh test, breakpoint never hit);

```

        public RemoteWebDriver GetDriver(DesiredCapabilities caps)
        {

            HttpCommandExecutor executor = new HttpCommandExecutor(
                new Uri("http://<grid_url>"), TimeSpan.FromSeconds(60), false);

            executor.BeforeRemoteHttpRequest += BeforeRemoteHttpRequestHandler;
            
            return new RemoteWebDriver(executor, caps);
        }

        private void BeforeRemoteHttpRequestHandler(
            object sender, BeforeRemoteHttpRequestEventArgs e)
        {
            // Note: WebRequest.DefaultWebProxy is an IWebProxy implementation.
            // This could be anything, from adding additional headers, to modifying
            // the content of the request. Use with extreme caution.

           //BREAKPOINT here never gets hit
            e.Request.Proxy = WebRequest.GetSystemWebProxy();
        }
```

**Comment by jimevans** [comment]
Is it possible you have some exception handling in the code that calls `GetDriver()`? There's a legit bug here, but not hitting the breakpoint isn't it.

**Comment by gschultz77** [comment]
Thanks for the quick reply. 
There’s no exception handling in the code that calls GetDriver().... what’s the legit bug you referred to?
Cheers.

**Comment by jimevans** [comment]
The bug is that the request can't be modified once the request body stream (if any) has been written to. That wouldn't affect whatever's happening in your environment. I can't reproduce what you're seeing, that the event isn't getting fired.

**Comment by gschultz77** [comment]
Ok thanks. Does that bug mean that the configuration I’ve tried to apply in App.Config is being ignored? And that the default system proxy is not being used? Or have I misunderstood?

I guess I’m trying to establish whether setting up the HttpCommandExecutor and the associated event handler is now the _only_ way to specify your proxy and ensure it’s being used when creating a RemoteWebDriver..... or not.

**Comment by jimevans** [comment]
In the currently released binaries, if you set the Proxy property within the event handler, an exception will be thrown. And no, the system proxy is not being used at present. Going forward, creating your own `HttpCommandExecutor` and using the event handler will be the only way to set a proxy for a `RemoteWebDriver` instance to use between the C# code and the Java remote server/grid.

The code you're using looks correct; I cannot speak to why your event handler isn't getting invoked. That does not happen for me.

**Comment by gschultz77** [comment]
This issue has now been resolved with the 3.12.0 release and ability to successfully use a HttpCommandExecutor and explicitly set the proxy in the BeforeRemoteHttpRequestHandler event handler.
Thanks :-)
